### PR TITLE
Updated to support Top and Bottom layout gravities.

### PR DESCRIPTION
**Pull request by ka05**
You can now make the banner display on the top and bottom and by default will remain at the top left corner as it has been. README.md has been updated with example details.

Please review A.S.A.P. Thanks.

### CODE REVIEW

**Comment by ka05** [comment]
Just checking to see if anyone has looked at my PR in hopes to merge and update the maven version. I Need to use this functionality for a client asap. Thanks!

File: README.md [hunk 1]
```diff
@@ -109,7 +109,11 @@ easylauncher {
 | `customColorRibbonFilter("label", "#DCDCDC")` | ![](icons/customColorRibbon2.png) |
 | `customColorRibbonFilter("label", "#DCDCDC", "#000000")` | ![](icons/customColorRibbon3.png) |
 | `overlayFilter(new File("example-custom/launcherOverlay/beta.png"))` | ![](icons/overlay.png) |
-
+| Custom Positioning | |
+| Options: `Default`, `BottomHorizontal`, `TopHorizontal`
```
Comment (akaita) [inline README.md]: No need for these two lines

File: example-custom/build.gradle [hunk 2]
```diff
@@ -56,7 +56,13 @@ easylauncher {
             // Add one more filter to all `qa` variants
             filters = redRibbonFilter()
         }
-        staging {}
+        staging {
+            filters = [
+                    customColorRibbonFilter("debug", "#556600CC", "#FFFFFF", "BottomHorizontal")
+//                    customColorRibbonFilter("debug", "#6600CC", "#FFFFFF", "TopHorizontal")
+//                    customColorRibbonFilter("debug", "#6600CC", "#FFFFFF")
```
Comment (akaita) [inline example-custom/build.gradle]: No need fo rthe commented lines

File: example-custom/src/main/res/values-v26/colors.xml [hunk 3]
```diff
@@ -1,4 +1,4 @@
 <?xml version="1.0" encoding="utf-8"?>
 <resources>
-    <color name="ic_launcher_background">#1c7917</color>
+    <color name="ic_launcher_background">#1CCCA5</color>
```
Comment (akaita) [inline example-custom/src/main/res/values-v26/colors.xml]: I don't really mind about it. I'm just curious. Why spend any time changing this?

File: plugin/src/main/groovy/com/akaita/android/easylauncher/filter/ColorRibbonFilter.java [hunk 4]
```diff
@@ -9,15 +11,21 @@
...
 
     static final boolean debug = Boolean.parseBoolean(System.getenv("EASYLAUNCHER_DEBUG"));
 
+    private Logger logger;
+
     final Color ribbonColor;
 
     final Color labelColor;
 
+    final LayoutPosition position;
```
Comment (akaita) [inline plugin/src/main/groovy/com/akaita/android/easylauncher/filter/ColorRibbonFilter.java]: `LabelPosition` sounds more meaningfull to me

File: plugin/src/main/groovy/com/akaita/android/easylauncher/filter/ColorRibbonFilter.java [hunk 5]
```diff
@@ -9,15 +11,21 @@
...
 import java.awt.geom.Rectangle2D;
 import java.awt.image.BufferedImage;
+import java.util.logging.Logger;
+
 
 public class ColorRibbonFilter implements EasyLauncherFilter {
 
     static final boolean debug = Boolean.parseBoolean(System.getenv("EASYLAUNCHER_DEBUG"));
 
+    private Logger logger;
```
Comment (akaita) [inline plugin/src/main/groovy/com/akaita/android/easylauncher/filter/ColorRibbonFilter.java]: Logger should die for this plugin

File: plugin/src/main/groovy/com/akaita/android/easylauncher/filter/ColorRibbonFilter.java [hunk 6]
```diff
@@ -86,15 +166,24 @@ public void apply(BufferedImage image) {
...
         }
         g.dispose();
     }
 
-    private Font getFont(int maxLabelWidth, FontRenderContext frc) {
-        int max = largeRibbon ? 64 : 32;
+    private Font getFont(int iconHeight, int maxLabelWidth, FontRenderContext frc) {
+//        int max = largeRibbon ? 64 : 32;
+//        int max = largeRibbon ? 45 : 32;
+        int max = iconHeight / 8;
```
Comment (akaita) [inline plugin/src/main/groovy/com/akaita/android/easylauncher/filter/ColorRibbonFilter.java]: This is actually something I mentioned in https://github.com/akaita/easylauncher-gradle-plugin/pull/24 . I think this is the right path to take

File: plugin/src/main/groovy/com/akaita/android/easylauncher/filter/ColorRibbonFilter.java [hunk 7]
```diff
@@ -62,21 +86,77 @@ public void apply(BufferedImage image) {
...
         FontRenderContext frc = new FontRenderContext(g.getTransform(), true, true);
+        // calculate the rectangle where the label is rendered
         int maxLabelWidth = calculateMaxLabelWidth(y);
-        g.setFont(getFont(maxLabelWidth, frc));
+        g.setFont(getFont(height, maxLabelWidth, frc));
+
         Rectangle2D labelBounds = g.getFont().getStringBounds(label == null ? "" : label, frc);
+        int labelHeight = (int) labelBounds.getHeight();
+
+        int padding = largeRibbon ? (labelHeight / 8) : (labelHeight / 10);
```
Comment (akaita) [inline plugin/src/main/groovy/com/akaita/android/easylauncher/filter/ColorRibbonFilter.java]: Tbh, this is probably a better way to handle adaptive icons than the current one 👍 . 
I just need to test different scenarios to see how it looks.

**Comment by akaita** [comment]
@ka05 sorry for the delay 🙇 . I was too busy and could do any OS last few months... Thanks for you patience.

The PR looks promising, improving more than it promises. 

I added a few comments. Feel free to address them. 
Otherwise I'll most probably merge and use it a base later on tonight.

I feel confident the next release will relay on this feature together with a couple others. Expect it to come in a matter of days.

**Comment by akaita** [comment]
I'll merge and tweak starting from you changes. 
I need to clean up the code a bit, make it more readable (including old stuff), fix support for adaptive icons, ...

Thanks so much, your PR is a great starting point!

**Pull request merged**